### ISSUE TITLE
Ability to rename model/field without losing data

### ISSUE THREAD

**Issue by flybayer** [aws-amplify/amplify-cli#781]
**Is your feature request related to a problem? Please describe.**
In almost every project, you eventually want to rename a model or model field because you later have a deeper understanding of the domain and how it should be modeled.

**Describe the solution you'd like**
I think [graph.cool's temporary `@rename` directive](https://www.graph.cool/docs/reference/database/data-modelling-eiroozae8u#temporary-directives) works very well.

```graphql
## Renaming the `Post` type to `Story`, and its `text` field to `content`
type Story @model @rename(oldName: "Post") {
  content: String @rename(oldName: "text")
}
```

After deploying the change, you have to manually remove the @rename directive.

**Describe alternatives you've considered**
None

**Comment by mikeparisstuff** [comment]
I agree that this is useful, but the pattern will have to be slightly different given the differences between DynamoDB and SQL. This method works for SQL as the respective rename operations boil down to `ALTER table RENAME TO` and `ALTER table RENAME COLUMN` but there is no equivalent in DynamoDB. It would be possible to create a similar `@alias` directive that would allow you to expose a different name for some underlying field.

```graphql
type Story @model {
  # data would be stored in DynamoDB as "text" and exposed via the API as "content"
  content: String @alias(name: "text")
}
```

We could use a similar pattern for table renames as well. E.G.

```graphql
type Story @model @alias(name: "Post") { ... }
```

With this approach, you would not be able to remove the @alias directive afterwards as there is no DDL statement being performed. The resolvers would handle the alias operation at runtime and you would continue to see the old name of the field in the DynamoDB table. 

I will record this as a feature request but also want to propose a different solution. In previous issues, we have discussed options that would allow users to load their own transformers as part of the CLI's compile step. This would allow users to write directives as Javascript classes that implement the transformer interface. User could then upload the transformer classes to npm or store them locally in the project and tell the amplify cli where to find them such that the logic is run during the compile step. This should unblock most use cases like this in the future and provide another mechanism through which you can customize your API.
